electron/src/components/charts/WordOrEmojiCountChart.tsx

```diff
@@ -10,30 +10,27 @@ interface WordOrEmojiCountProps {
   db: sqlite3.Database;
   titleText: string;
   labelText: string;
-  isEmoji: boolean;
+  filters: WordOrEmojiTypes.Filters;
   colorInterpolationFunc: (t: number) => string;
 }
 
 export default function WordOrEmojiCountChart(props: WordOrEmojiCountProps) {
-  const { db, colorInterpolationFunc, titleText, labelText, isEmoji } = props;
+  const { db, colorInterpolationFunc, titleText, labelText, filters } = props;
   const [words, setWords] = useState<string[]>([]);
   const [count, setCount] = useState<number[]>([]);
 
   useEffect(() => {
     async function fetchWordData() {
       try {
-        const data = await chatBro.queryEmojiOrWordCounts(db, {
-          isEmoji,
-          isFromMe: true,
-        });
+        const data = await chatBro.queryEmojiOrWordCounts(db, filters);
         setWords(data.map((obj) => obj.word));
         setCount(data.map((obj) => obj.count));
       } catch (err) {
         log.error(`ERROR fetching for component for ${titleText}`, err);
       }
     }
     fetchWordData();
-  }, []);
+  }, [db, titleText, filters]);
```

Comment:
what does this do

Reply:
https://dev.to/rozenmd/understanding-useeffect-the-dependency-array-obg

^ that will explain it better than i ever could